shop/src/main/java/ro/msg/learning/shop/service/strategy/SingleLocationStrategy.java

```diff
@@ -0,0 +1,48 @@
+package ro.msg.learning.shop.service.strategy;
+
+import ro.msg.learning.shop.model.Location;
+import ro.msg.learning.shop.model.OrderDetail;
+import ro.msg.learning.shop.model.Stock;
+import ro.msg.learning.shop.repository.LocationRepository;
+import ro.msg.learning.shop.repository.StockRepository;
+import ro.msg.learning.shop.service.exceptions.NotFoundException;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class SingleLocationStrategy implements LocationStrategy{
+
+    private final StockRepository stockRepository;
```

Comment:
You should inject at this level the services and not the repositories of the entities. The only layer that is communicating directly with the repositories is the service layer. Same applies for the other strategy implementation